### DISCUSSION TITLE
Visually illustrate model architectures based on config

### DISCUSSION THREAD

**Discussion by NixBiks** [explosion/spaCy#9016] [New Features & Project Ideas]
Have you considered adding a tool that can illustrate model architectures visually based on `config.cfg`?

I haven't given it much thought about how yet - I wanted to hear your thoughts about it first.

A graph can be a lot nicer to read / easier to understand as model documentation than a config file.

**Comment by polm** [comment]
Thanks for the suggestion! We have discussed this internally before, with the idea of generating something like the [diagrams here](https://spacy.io/usage/embeddings-transformers#embedding-layers), but we aren't working on it at the moment. Any suggestions would be welcome.

![pipeline diagram](https://spacy.io/tok2vec-listener-8c4d53807708b270c07a085f4a2da75f.svg)
